doc/rate-limiting.md

```diff
@@ -253,6 +253,87 @@ Check out the following user guides for examples of rate limiting services with
 * One HTTPRoute can only be targeted by one RLP.
 * One Gateway can only be targeted by one RLP.
 * RLPs can only target HTTPRoutes/Gateways defined within the same namespace of the RLP.
+* 2+ RateLimitPolicies cannot target network resources that define/inherit the same exact hostname.
+
+#### Limitation: Multiple network resources with identical hostnames
+
+Kuadrant currently does not support multiple RateLimitPolicies simultaneously targeting network resources that declare identical hostnames. This includes multiple HTTPRoutes that specify the same hostnames in the `spec.hostnames` field, as well as HTTPRoutes that specify a hostname that is identical to a hostname specified in a listener of one of the route's parent gateways or HTTPRoutes that don't specify any hostname at all thus inheriting the hostnames from the parent gateways. In any of these cases, **a maximum of one RateLimitPolicy targeting any of those resources that specify identical hostnames is allowed**.
+
+Moreover, having **multiple resources that declare identical hostnames** may lead to unexpected behavior and therefore **should be avoided**.
+
+This limitation is rooted at the underlying components configured by Kuadrant for the implementation of its polcies and the lack of information in the data plane regarding the exact route that honored by the API gateway in cases of conflicting hostnames.
```

Comment:
same as above: polcies typo and "that honored by" question which I am not sure about